wpilibc/src/main/native/include/frc/ResistanceCalculator.h

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include <wpi/circular_buffer.h>
+
+namespace frc {
+
+/**
+ * Finds the resistance of a channel or the entire robot using a running linear
+ * regression over a window. Must be updated with current and voltage
+ * periodically using the <code>Calculate</code> method. <p>To use this for
+ * finding the resistance of a channel, use the calculate method with the
+ * battery voltage minus the voltage at the motor controller or whatever is
+ * plugged in to the PDP at that channel.</p>
+ */
+class ResistanceCalculator {
+ public:
+  static constexpr int kDefaultBufferSize = 250;
+  static constexpr double kDefaultRSquaredThreshold = 0.75;
+
+  /**
+   * Create a ResistanceCalculator to find the resistance of a channel using a
+   * running linear regression over a window. Must be updated with current and
+   * voltage periodically using the <code>Calculate</code> method.
+   *
+   * @param bufferSize The maximum number of points to take the linear
+   * regression over.
+   * @param krSquaredThreshold The minimum R² value (0 to 1) considered
+   * significant enough to return the regression slope instead of NaN. A lower
+   * threshold allows resistance to be returned even with noisier data.
+   */
+  ResistanceCalculator(int bufferSize, double krSquaredThreshold);
+
+  /**
+   * Create a ResistanceCalculator to find the resistance of a channel using a
+   * running linear regression over a window. Must be updated with current and
+   * voltage periodically using the <code>Calculate</code> method. <p>Uses a
+   * buffer size of 250 and an R² threshold of 0.5.</p>
+   */
+  ResistanceCalculator();
+
+  ~ResistanceCalculator() = default;
+  ResistanceCalculator(ResistanceCalculator&&) = default;
+  ResistanceCalculator& operator=(ResistanceCalculator&&) = default;
+
+  /**
+   * Update the buffers with new (current, voltage) points, and remove old
+   * points if necessary.
+   *
+   * @param current The current current
+   * @param voltage The current voltage
+   * @return The current resistance in ohms
+   */
+  double Calculate(double current, double voltage);
```

Comment:
```suggestion
  units::ohm_t Calculate(units::ampere_t current, units::volt_t voltage);
```
From `<units/imedance.h>`, `<units/current.h>`, and `<units/voltage.h>` respectively.